ci/gpu/build.sh

```diff
@@ -37,6 +37,7 @@ logger "Activate conda env..."
 source activate gdf
 conda install -c rapidsai/label/cuda${CUDA_REL} -c rapidsai-nightly/label/cuda${CUDA_REL} cudf=${CUDF_VERSION} rmm=${RMM_VERSION} nvstrings=${NVSTRINGS_VERSION}
 conda install -c conda-forge lapack cmake==3.14.3
+conda install -c teju85 libclang 
```

Comment:
Why is this package in a personal channel? 

Additionally, there is only a py37 version, so this command forcibly updates the python in the environment to 3.7.

Reply:
Because I don't have access to upload packages to the official rapidsai channel. However, I have created recipe for building this here: https://github.com/teju85/clang-recipe. I can even provide py35 package too.

If you can help upload this package to any of our official channels, that'd be fine by me too.

Reply:
I have fixed this now and made sure that the `libclang` conda package is of type noarch. What do you think now, @raydouglass ?

Reply:
@teju85 Please open an ops issue to get this conda package into our official channel.

Reply:
Could we add this dependency to the `third_party` subdirectory in cuML so we don't have to fetch it separately? It seems small enough to add. (I think this was Dante's suggestion IIRC)

Reply:
I assume you mean as a git submodule? I'm not sure I like that option. Other cuDF libraries are likely to follow suit on clang-format, so duplicating the dependency in a submodule of every library on the user's machine seems overkill -- if it's installed via conda in their environment then all libraries can access it. Also, it has been suggested (by Rory for XGBoost) that any sub-dependency is too much when trying to get XGBoost to adopt dependencies -- I'm specifically thinking about RMM, here. (in other words, for some users, our modules are more adoptable if they don't have hierarchical dependencies...)

Reply:
John, I think it's easier to manage this as a dependency via conda itself.

Reply:
If this package makes it to the official channel, then I'm cool with it....  Could we rename it to clang-format though instead of libclang?

This introduces a dependency on conda in our build process though (I think Dante pointed this out. Previously, a developer could build from source without conda installed, and I think that's a nice property to maintain. I guess two possible workarounds would be to document installation of the libclang package via some other method or to allow a cmake flag or something that allows developers to disable the libclang checks if the tool is not present... Maybe some other option?

Reply:
If one doesn't/can't use conda, then they can install the exact clang-format `8.0.0` version using any of their favorite package management SW like for eg. apt-get. This can certainly be documented, if we have a readme or something for non-conda developers.

Reply:
this can go both in the `BUILD.md` and the `cpp/README.md` docs

Reply:
I had updated `BUILD.md`. Now, updated `cpp/README.md` as well.

Reply:
> If one doesn't/can't use conda, then they can install the exact clang-format 8.0.0 version using any of their favorite package management SW like for eg. apt-get. 

This is, unfortunately, untrue in Centos7, as the requirement here is on Clang 8 and I don't see an easy option to install that through yum.

Reply:
can you atleast try to download the tarball from conda package and see if it works? I don't have access to centos machines to confirm if atleast this tarball works for folks whose pkg-mgr doesn't yet have 8.0.0.

Reply:
Certainly, I can give it a try. Stay tuned...

Reply:
I did have a problem building your `libclang` from source. I suspect it's related to the need for `devtoolset` packages to upgrade gcc in order to build cuML.

@teju85, with the link you've provided for the conda tarball, I was able to verify this does work in Centos7. I forced a failure just to be sure:

```
clang-format failed! Run 'diff -y /share/workspace/cuml/cpp/src/knn/knn.h /home/cjnolet/workspace/cuml/cpp/build/clang-format/src/knn/knn.h | more' to know more about the formatting violations!
Clang-format failed! Look at errors above and fix them, you 
can also run the following command to bulk fix all the flagged
 violations above:  make fix-format
make[2]: *** [CMakeFiles/format] Error 255
make[1]: *** [CMakeFiles/format.dir/all] Error 2
make: *** [all] Error 2
```

Reply:
Thanks @cjnolet for verifying this.